src/authenticator.rs

```diff
@@ -1105,10 +1109,14 @@ where
             return Err(Status::SecurityStatusNotSatisfied);
         }
 
+        self.state.runtime.encryption_key = None;
```

Comment:
Here you probably want to have something like 

```rust
if let Some(key) = self.state.runtime.encryption_key.take() {
    syscall!(self.trussed.delete(key));
}
```

Reply:
Yup. Assumed the key is cached on the backend side. Applied as suggested.